test/test_tensordict.py

```diff
@@ -539,6 +539,29 @@ def test_stacked_td(stack_dim, device):
     assert (std5.contiguous() == sub_td.contiguous().unbind(1)[0]).all()
 
 
+@pytest.mark.parametrize("device", get_available_devices())
+@pytest.mark.parametrize("stack_dim", [0, 1])
+def test_stacked_indexing(device, stack_dim):
```

Comment:
One way to do this is to create the stacked td like this

```
tensordict = torch.stack(list(tensordict.unbind(stacked_dim)), stacked_dim)
```
like this you can stack along any dim and all of your tensordicts will have the same shape, hence you can check that the behaviour of indexing is the same in every case

Reply:
Neat, I've updated the test and used this pattern.